kvstorage: don't write the cluster version key in new versions

RaduBerinde opened the pull request:
Stop writing the cluster version key if the cluster version is new enough. This will allow us to completely remove any knowledge of this key during 23.2 store initialization code (e.g. in `assertEnginesEmpty`).

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/97743)
<!-- Reviewable:end -->

RaduBerinde commented:
Gentle ping. Thanks!

jbowens reviewed (approved):
sorry for the delay, <img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/> 

> Reviewed 6 of 6 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/97743#-:-NPd4zfv0qjrQsjj9p8F:blbpcdz)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @nicktrav)



<!-- Sent from Reviewable.io -->

RaduBerinde commented:
TFTR!

RaduBerinde committed:
kvstorage: don't write the cluster version key in new versions

Stop writing the cluster version key if the cluster version is new
enough. This will allow us to completely remove any knowledge of this
key during 23.2 store initialization code (e.g. in
`assertEnginesEmpty`).

Epic: none
Release note: None

RaduBerinde commented:
bors r+

RaduBerinde commented:
bors r+
